radix-engine/src/engine/call_frame.rs

```diff
@@ -58,8 +59,10 @@ pub struct CallFrame<
 
     /// Referenced values
     refed_values: HashMap<StoredValueId, ValueRefType>,
-    // TODO: Remove
-    refed_components: HashMap<ComponentAddress, Component>,
+
+    // TODO: Merge with refed_values
+    /// Readable values
+    readable_values: HashSet<Address>,
```

Comment:
I can no longer reason about the correctness of the implementation in this file by just looking at the diff, especially with so many terms. Strongly need documentation about our assumptions, abstractions and algorithms.